### ISSUE TITLE
"permission denied" errors in output

### ISSUE THREAD

**Issue by fhemberger** [muesli/duf#1]
Just installed `duf-r17.3818846-1` from AUR and ran it as regular user: 

![screenshot of `duf` output](https://user-images.githubusercontent.com/153481/93732852-76eaec00-fbd3-11ea-9314-a0341a0840a6.png)

Running with `sudo`, the "permission denied" errors are gone, but it should work with regular users as well.

**Comment by muesli** [comment]
I assume you'll get a similar output when running the regular `df` as non-root? I agree though, we need to annotate the errors, so the user knows what permission actually was denied.

**Comment by fhemberger** [comment]
No, `df` works fine without errors for me. Anything I can do to help you debug this?

**Comment by muesli** [comment]
I've just annotated the errors, so we should get a bit more context. Can you retry with latest git-master?

**Comment by fhemberger** [comment]
Ah, it's Docker: `/var/lib/docker/overlay2/…/merged` and `/run/docker/netns/…`.

**Comment by muesli** [comment]
It's weird this doesn't produce any permission errors in `df`. I need to check why they decide to skip this particular mount point.

**Comment by muesli** [comment]
Can you grep the corresponding lines to `/var/lib/docker/overlay2/…/merged` from `/proc/self/mountinfo`, please?

**Comment by fhemberger** [comment]
Maybe they just skip overlay/nsfs altogether?

```
568 28 0:57 / /var/lib/docker/overlay2/…/merged rw,relatime shared:327 - overlay overlay rw,lowerdir=…,upperdir=…,workdir=…,index=off,xino=off

1410 26 0:4 net:[4026532618] /run/docker/netns/… rw shared:397 - nsfs nsfs rw
```

**Comment by pvaret** [comment]
I'm also affected by this, but with a different error message:

```
/sys/kernel/debug/tracing: permission denied
```

I'm not sur what duf is attempting here. That directory is the `tracefs` mount point, I don't know what duf needs to know about it beyond that.

**Comment by muesli** [comment]
@pvaret What `duf` does here is a `statfs` call to retrieve things like block-size, available/used blocks etc.

**Comment by mamyn0va** [comment]
Same here:

```
❯ duf                                                                                                                                                                                                           49.3s
/run/user/1000/doc: operation not permitted
/var/lib/docker/overlay2/d8654af827b462362452b297ee0153105505961d592559905f7740b0bb10ea7d/merged: permission denied
/run/docker/netns/2421352ad105: permission denied
╭──────────────────────────────────────────────────────────────────────────────────────────────╮
│ 1 local devices                                                                              │
├────────────┬────────┬────────┬───────┬──────────────────────────────┬───────────┬────────────┤
│ MOUNTED ON │   SIZE │   USED │ AVAIL │ USE%                         │ TYPE      │ FILESYSTEM │
├────────────┼────────┼────────┼───────┼──────────────────────────────┼───────────┼────────────┤
│ /          │ 216.4G │ 155.9G │ 49.4G │ [##############......] 72.0% │ ext2/ext3 │ /dev/sda2  │
╰────────────┴────────┴────────┴───────┴──────────────────────────────┴───────────┴────────────╯
╭────────────────────────────────────────────────────────────────────────────────────────────╮
│ 6 special devices                                                                          │
├────────────────┬──────┬────────┬───────┬──────────────────────────────┬───────┬────────────┤
│ MOUNTED ON     │ SIZE │   USED │ AVAIL │ USE%                         │ TYPE  │ FILESYSTEM │
├────────────────┼──────┼────────┼───────┼──────────────────────────────┼───────┼────────────┤
│ /dev           │ 7.8G │     0B │  7.8G │ [....................] 0.0%  │ tmpfs │ dev        │
│ /run           │ 7.8G │   1.7M │  7.8G │ [....................] 0.0%  │ tmpfs │ run        │
│ /dev/shm       │ 7.8G │ 267.3M │  7.5G │ [....................] 3.4%  │ tmpfs │ tmpfs      │
│ /sys/fs/cgroup │ 4.0M │     0B │  4.0M │ [....................] 0.0%  │ tmpfs │ tmpfs      │
│ /tmp           │ 7.8G │ 850.9M │  6.9G │ [##..................] 10.7% │ tmpfs │ tmpfs      │
│ /run/user/1000 │ 1.6G │  92.0K │  1.6G │ [....................] 0.0%  │ tmpfs │ tmpfs      │
╰────────────────┴──────┴────────┴───────┴──────────────────────────────┴───────┴────────────╯
```

**Comment by pvaret** [comment]
@muesli Interesting! Perhaps duf should handle gracefully special filesystems for which those numbers don't make a lot of sense.

**Comment by muesli** [comment]
Thanks for all the feedback! I think it makes sense to ignore permission errors in `duf` entirely, as they're not an unexpected error.

I've implemented this in #14. More info in the PR.
